src/lib/transaction.ts

```diff
@@ -1,26 +1,25 @@
 import { DecodedTx, Key, KeyType, Path, Recipient, TxOut, UTXO } from '../types/domain'
-import { TransactionBuilder } from 'bgoldjs-lib';
-import { GetKeyBackendResponse, GetWalletBackendResponse, ListUnspentsBackendResponse } from 'response';
-import BigNumber from "bignumber.js";
+import { TransactionBuilder } from 'bgoldjs-lib'
+import { GetKeyBackendResponse, GetWalletBackendResponse, ListUnspentsBackendResponse } from 'response'
+import BigNumber from 'bignumber.js'
 import { btcToSatoshi, satoshiToBtc } from './utils/helpers'
-import { decrypt } from './crypto';
-import { API_ERROR } from './constants';
+import { decrypt } from './crypto'
+import { API_ERROR } from './constants'
 import { KeyModule } from './key'
 import { WalletApi } from './wallet'
-import { CurrencyBackendApi } from './backend-api';
-import { BitcoinOperations } from './bitcoin-operations';
+import { CurrencyBackendApi } from './backend-api'
+import { BitcoinOperations } from './bitcoin-operations'
 
 export interface TransactionApi {
-  send(userToken: string, walletId: string, recipients: Recipient[], xprv?: string, passphrase?: string): Promise<string>
+  send(userToken: string, walletId: string, recipients: Recipient[], xprv?: string, passphrase?: string, feeRate?: number): Promise<string>
 }
 
 export const transactionApiFactory = (backendApi: CurrencyBackendApi, keyModule: KeyModule, bitcoin: BitcoinOperations, walletApi: WalletApi): TransactionApi => {
 
   const send = async (
-    userToken: string, walletId: string, recipients: Recipient[], xprv?: string, passphrase?: string
-  ): Promise<string> => {
-    const { recommended } = await backendApi.getFeesRates()
-    const unspentsResponse = await walletApi.listUnspents(userToken, walletId, recommended, recipients)
+    userToken: string, walletId: string, recipients: Recipient[], xprv?: string, passphrase?: string, userProvidedFeeRate?: number): Promise<string> => {
+    const feeRate = userProvidedFeeRate || (await backendApi.getFeesRates()).recommended
```

Comment:
We're fetching recommended backendApi feeRate in order to send it back to the backend in next line?
Is it wise communication pattern?
What about making this param optional and read recommended rate on the backend side in case of absence?

Reply:
poprawione